src/rustbook/javascript.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// The rust-book JavaScript in string form.
+
+pub static JAVASCRIPT: &'static str = r#"
+<script type="text/javascript">
+document.addEventListener("DOMContentLoaded", function(event) {
+  document.getElementById("toggle-nav").onclick = toggleNav;
+  function toggleNav() {
+    var toc = document.getElementById("toc");
+    var pagewrapper = document.getElementById("page-wrapper");
+    if ( "block" === toc.style.display ) {
+      toc.style.display = "none";
+      pagewrapper.style.display = "block";
+    } else {
+      toc.style.display = "block";
+      pagewrapper.style.display = "none";
+    }
+  };
+});
+</script>
+"#;
```

Comment:
I borrowed using a module from the way the CSS is implemented: 
https://github.com/rust-lang/rust/blob/master/src/rustbook/css.rs

I agree it is an "interesting" approach, but since it's my first pull request I tried to stick to what I thought was convention.